internal/kafka/internal/workers/kafka_mgrs/promotion/chain/reconciler_chain.go

```diff
@@ -0,0 +1,69 @@
+package chain
+
+import (
+	"github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/internal/kafka/internal/api/dbapi"
+)
+
+// ReconcileAction - This is the interface that must be implemented by each action that the reconciler will execute.
+// T - is the type of result the action will generate
+type ReconcileAction[T any] interface {
+	// PerformJob - This is the method where the Action implementation will perform the real job
+	// * kafkaRequest - the kafka request to reconcile
+	// * currentResult - object containing the results that have been produced up to this point into the pipeline
+	// Return values
+	// * res Result[T] - the result produced by this action
+	// * finished bool - returning `true` here will interrupt the pipeline execution
+	// * err error - an eventual error. Returning an error always interrupts the pipeline execution
+	PerformJob(kafkaRequest *dbapi.KafkaRequest, currentResult Result[T]) (res Result[T], finished bool, err error)
+}
+
+// Result - This struct contains the result produced by each action
+type Result[T any] struct {
+	hasValue bool
+	value    T
+}
+
+func (r *Result[T]) SetValue(value T) {
+	r.value = value
+	r.hasValue = true
+}
+
+func (r *Result[T]) Value() T {
+	return r.value
+}
+
+func (r *Result[T]) HasValue() bool {
+	return r.hasValue
+}
+
+// The ReconcileActionRunner is the engine of the action chain. It runs each action one after the other
```

Comment:
The `The` word should be removed. Idiomatic Go documentation contains the name of the method / type to be documented as the first word in the document comment.

Reply:
An optional leading article is admitted AFAIK

Reply:
My comment is based on information seen around godoc's approach to documentation, https://go.dev/blog/godoc, where beginning with the name of the element it describes seems to be the convention: 
```
Notice this comment is a complete sentence that begins with the name of the element it describes. This important convention allows us to generate documentation in a variety of formats, from plain text to HTML to UNIX man pages, and makes it read better when tools truncate it for brevity, such as when they extract the first line or sentence.
```

Reply:
Look at [here](https://tip.golang.org/doc/comment):
```
package bytes

// A Buffer is a variable-sized buffer of bytes with Read and Write methods.
// The zero value for Buffer is an empty buffer ready to use.
type Buffer struct {
    ...
}
```

```
package io

// A LimitedReader reads from R but limits the amount of
// data returned to just N bytes. Each call to Read
// updates N to reflect the new amount remaining.
// Read returns EOF when N <= 0.
type LimitedReader struct {
    R   Reader // underlying reader
    N   int64  // max bytes remaining
}
```

and many others.

Reply:
However, the usage of ` - ` does not seem that conventional . For example, in :
`// PerformJob - ...`

Reply:
Yes. I'm going through the code and removing all of them.

Reply:
I still see some instances. For example:
```
	// PerformJob - This is the method where the Action implementation will perform the real job
```